packages/cli/src/commands/setup/auth/templates/auth.ts.template

```diff
@@ -75,7 +75,7 @@ export const hasRole = (roles: AllowedRoles): boolean => {
     return false
   }
 
-  const currentUserRoles = context.currentUser?.roles
+  const currentUserRoles = context?.currentUser?.roles
```

Comment:
I don't think we need the `?` after context, as its always there?

Reply:
Actually not my idea, it was already there before, but inconsistently:

https://github.com/redwoodjs/redwood/blob/8e116c37fc68ef75244c26b8771791973c0f88f5/packages/cli/src/commands/setup/auth/templates/dbAuth.auth.ts.template#L77

Also the graphQL root schema doesn't trust the context to always be defined:

https://github.com/redwoodjs/redwood/blob/8e116c37fc68ef75244c26b8771791973c0f88f5/packages/graphql-server/src/rootSchema.ts#L80

Counter-Point: What's the damage in having it in there? On the upside, in case it should ever not be defined, the code will fail gracefully.

Reply:
It's probably just legacy code, no need to change it in other places now though!

Context in your redwood app (not inside the framework or other places like graphql) - should always be defined, even if as an empty object.

Reply:
> Counter-Point: What's the damage in having it in there?

You didn't answer my question though :wink: 

But so what – i changed it now as requested :+1:

Reply:
> You didn't answer my question though 😉

Because writing code that doesn't do anything seems odd to me. By the same logic we could just wrap everything in 

```js
if (true) {
// run your actual code 
}
```
Technically it does no harm... but why add these changes if they have no benefit?

Reply:
I see your point. That is of course only as long as `true` really is `true` – so that may have been a good fallback in case `context` ever should not be defined, but on the other hand that may be intended to run into a hard error in that case b/c we would like to know about it.

This example reminds me of a javascript fun-fact: Did you know [`undefined` is not necessarily always `undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#description)? JS … the good parts :rofl: 

Anyway, kindly merge if you're happy with this PR now :pray: